### ISSUE TITLE
UI unresponsive - Changing label and archiving immediately after doesn't remove email from inbox

### ISSUE THREAD

**Issue by bellebethcooper** [nylas/N1#1385]
On Mac El Capitan, running Nylas N1 Version 0.4.5-b0ec335 (0.4.5-b0ec335) with a single Gmail account with one alias.

If I hit "L" and choose a label, then immediately hit "E" to archive, the inbox label disappears, sometimes the email disappears from the inbox then pops back into the inbox (sometimes the message itself doesn't move at all), then the inbox label pops back onto the message. I can do this multiple times in a row and it keeps happening.

If I click into a different folder and come back, the message is no longer in the inbox.

Checking the dev activity window, there are no errors. The label change seems to be completing correctly every time.

If I wait a couple of seconds after adding the label before I hit "E" to archive, the archive works as expected.

**Comment by orky** [comment]
+1

**Comment by lleger** [comment]
I am experiencing a similar issue. After I archive an email, it takes a while before the email is removed from the inbox view.

**Comment by twomoredays** [comment]
I'm on version Version 0.4.10-4e3595b and  El Capitan 10.11.3 and experiencing the same bug as @lleger, where marking an item as archived (by shortcut or click) does not remove it from the inbox for an interval of up to a few minutes.

**Comment by jstejada** [comment]
hi @twomoredays , @lleger, if you open up the dev activity window `(cmd/ctrl-alt-w)`, do you see the requests failing or going through? If you change views after archiving and come back, is it still in the inbox? Is it possible that you were trying to archive during inital sync and the action got queued up? Otherwise think your specific issue might actually be related to #781.

@bellebethcooper I'm going to try to reproduce the exact issue you are having

thanks!

**Comment by lleger** [comment]
> if you open up the dev activity window (cmd/ctrl-alt-w), do you see the requests failing or going through?

Yes, it shows what I assume is the correct request:
```
curl -X PUT -d '{"labels":["xxx123","abc123"]}' "https://xxx123:@api.nylas.com/threads/abc123"
```

> If you change views immediately after archiving and come back, is it still in the inbox or has been moved correctly?

No, but if I close and reopen the app it's gone.



> Is it possible that you were trying to archive during inital sync and the action got queued up?

I would not think so. It seems instead an issue with updating the UI.

**Comment by jstejada** [comment]
thanks for the details @lleger , we'll look into it.

**Comment by jstejada** [comment]
Hey folks—this should be fixed in 0.4.32, which shipped monday. Cheers!
